types/index.d.ts

```diff
@@ -127,13 +128,14 @@ declare module 'mongoose' {
    */
   export function isObjectIdOrHexString(v: any): boolean;
 
-  export function model<T>(name: string, schema?: Schema<T, any, any> | Schema<T & Document, any, any>, collection?: string, options?: CompileModelOptions): Model<T>;
-  export function model<T, U, TQueryHelpers = {}>(
+export function model<T, U = unknown, TQueryHelpers = {}, TSchema = any>(
```

Comment:
Dont you want to initialize the generic type TSchema with something else than any?

Reply:
I would prefer unknown, but that have side effect on some tests, but it works fine with any.